addons/web/static/src/js/views/abstract_view.js

```diff
@@ -183,84 +179,140 @@ var AbstractView = Factory.extend({
             this._updateMVCParams(params.searchQuery);
         }
 
+        this.withControlPanel = this.withControlPanel && params.withControlPanel;
+        this.withSearchPanel = this.withSearchPanel &&
+            this.multi_record && params.withSearchPanel &&
+            !('search_panel' in params.context && !params.search_panel);
+
+        if (this.withControlPanel || this.withSearchPanel) {
+            const { arch, fields, favoriteFilters } = params.controlPanelFieldsView || {};
+            const {
+                'control-panel': controlPanelInfo,
```

Comment:
is it good? see ref to ControlPanelComponent.modelExtension below
Are we going far enough?
I think we should not deconstruct that way.
Also it is maybe weird to have to extractArchInfo for all extensions instead of just the one we want... -> might give part of the solution to previous point

Reply:
I don't think so : it is rather similar to the principle of `get` and `dispatch` ; concerned models will do what they want with the archs info